tests/test_cluster_utils.py

```diff
@@ -21,26 +22,86 @@ def finalizer():
+    mon_count = self.cluster_obj.mon_count
+    log.info(f"Mon count before add = {mon_count}")
+    self.cluster_obj.scan_cluster()
+    self.cluster_obj.cluster.reload()
+    self.cluster_obj.cluster.data['spec']['mon']['allowMultiplePerNode'] = True
+    self.cluster_obj.cluster.apply(**self.cluster_obj.cluster.data)
+    yield
+    self.cluster_obj.mon_change_count(mon_count)
+    if mon_count != self.cluster_obj.mon_count:
+        log.error(f"Mon teardown failure")
+        log.error(
+            f"Expected: {mon_count}",
+            f"but found {self.cluster_obj.mon_count}"
+        )
+    log.info("Removed mon")
+    self.cluster_obj.cluster.data['spec']['mon'][
+        'allowMultiplePerNode'
+    ] = False
+    self.cluster_obj.cluster.apply(**self.cluster_obj.cluster.data)
+
+
+@pytest.fixture
+def mds_resource(request):
+    """
+    A fixture to handle mds resource cleanup
+    This function brings mds count to what it was before test started
+    """
+    self = request.node.cls
+    we_created_fs = False
+    if not self.cluster_obj.cephfs:
+        # cephfs doesn't exist , create one for this test
+        assert helpers.create_cephfilesystem()
+        self.cluster_obj.scan_cluster()
+        assert self.cluster_obj.cephfs
+        we_created_fs = True
+    mds_count = int(self.cluster_obj.mds_count / 2)
+    yield
+    self.cluster_obj.mds_change_count(mds_count)
+    current_count = int(self.cluster_obj.mds_count / 2)
+    if mds_count != current_count:
+        log.error(f"MDS teardown failure")
+        log.error(f"Expected: {mds_count} but found {current_count}")
+    if we_created_fs:
+        self.cluster_obj.cephfs.delete()
+        self.cluster_obj.cephfs = None
+
+
+@pytest.fixture
+def user_resource(request):
+    """
+    A fixture for creating user for test and cleaning up after test is done
+    """
+    self = request.node.cls
+    log.info("Creating user")
+    assert self.cluster_obj.create_user(self.username, self.caps)
+    yield
+    del_cmd = f"ceph auth del {self.username}"
+    log.info("User deleted")
+    self.cluster_obj.toolbox.exec_ceph_cmd(del_cmd)
+
+
 def setup(self):
     """
-    Setting up the environment for the test
+    Create CephCluster object to be consumed by tests
     """
-    self.cluster = CephCluster()
-    assert self.cluster.create_user(self.username, self.caps)
+    self.cluster_obj = CephCluster()
 
 
 def teardown(self):
     """
-    Tearing down the environment
+    Make sure at the end cluster is in HEALTH_OK state
     """
-    new_count = self.cluster.mon_count - 1
-    self.cluster.mon_change_count(new_count)
-    assert new_count == self.cluster.mon_count
-    new_mdscount = int(self.cluster.mds_count / 2) - 1
-    self.cluster.mds_change_count(new_mdscount)
-    assert new_mdscount * 2 == self.cluster.mds_count
-    del_cmd = f"ceph auth del {self.username}"
-    self.cluster.toolbox.exec_ceph_cmd(del_cmd)
+    assert self.cluster_obj.cluster_health_check(timeout=1200)
```

Comment:
Is 1200 the right value here? Just verifying as that's a 20 minute timeout.

Reply:
This is almost end of the test, for some reason cluster was taking time to reach HEALTH_OK condition. Since this is test I thought 1200 would be ok but it will come back in like 10 mins or so. But default timeout scaling we do (num_pods * 10) was not sufficient atleast in few runs of mine. Let me know if that needs to be changed.